### PR TITLE
feat: upgrade to CakePHP 4.4

### PR DESCRIPTION
**Pull request by batopa**
This PR needs to upgrade to CakePHP 4.4

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,21 +22,23 @@
     ],
     "require": {
         "php": "^7.4 || ^8.0",
-        "cakephp/cakephp": "~4.3.8",
-        "guzzlehttp/psr7": "^2.2.1",
+        "cakephp/cakephp": "~4.4.1",
         "cakephp/plugin-installer": "^1.3.1",
+        "guzzlehttp/psr7": "^2.2.1",
+        "mobiledetect/mobiledetectlib": "^2.8",
```
Comment (stefanorosanelli) [inline composer.json]: are we still using `mobiledetect` in BE5?

**Review by stefanorosanelli — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (batopa) [reply composer.json]: I'm going to remove it.

File: plugins/BEdita/Core/tests/TestCase/Command/TreeCheckCommandTest.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function setUp(): void
         parent::setUp();
 
         $this->useCommandRunner();
-        $this->loadModel('Trees');
+        $this->fetchTable('Trees');
```
Comment (stefanorosanelli) [inline plugins/BEdita/Core/tests/TestCase/Command/TreeCheckCommandTest.php]: it should be:

```php
$this->Trees = $this->fetchTable('Trees');
```

File: plugins/BEdita/Core/tests/TestCase/Command/TreeRecoverCommandTest.php [hunk 3]
```diff
@@ -53,7 +53,7 @@ public function setUp(): void
         parent::setUp();
 
         $this->useCommandRunner();
-        $this->loadModel('Trees');
+        $this->fetchTable('Trees');
```
Comment (stefanorosanelli) [inline plugins/BEdita/Core/tests/TestCase/Command/TreeRecoverCommandTest.php]: it should be:

```php
$this->Trees = $this->fetchTable('Trees');
```

**Review by stefanorosanelli — CHANGES_REQUESTED** [review]

**Review by stefanorosanelli — APPROVED** [review]

**Pull request merged**